woocommerce/woocommerce
# Convert notices to use native nonce generation from WC Admin

### All Submissions:






### Changes proposed in this Pull Request:



So the original issue where people could not dismiss admin notices (i.e. database update notice) was raised here https://github.com/woocommerce/woocommerce/issues/27424

In summary it was due to the fact that stored nonces in the db gets invalidated after a period of time making the notice not dismissible anymore.

Since at the time WC Admin did not have a native way to handle nonces, a work around was proposed and used here https://github.com/woocommerce/woocommerce/pull/28500

WC Admin will eventually support native way to accept and create nonces. Work being done here https://github.com/woocommerce/woocommerce-admin/pull/6726

So this PR is an update to remove the work around solution to utilize the WC Admin's native way of creating nonces.

Thanks @adrianduffell 

### How to test the changes in this Pull Request:

Activate WC Admin as a standalone plugin and checkout this PR https://github.com/woocommerce/woocommerce-admin/pull/6726 along with the one you're reviewing now.

1. Set your WooCommerce database version to something older like (4.9.0) You can find this in the `wp_options` table under `woocommerce_db_version` option name.
2. Reload your WooCommerce dashboard page and you should now see the database update notice.
3. Ensure you're able to click the update button and see that it starts progressing with the new notice.
4. After a small period of time, reload the page and ensure you see the notice that implies the update was done.
5. Click on the `Thanks` button and ensure it dismisses the notice.

### Other information:

This PR should only go in after WC Admin package with this change has been merged.




### Changelog entry

> Tweak - Use WC Admin's native notice nonce generation.
